crate/operator/create.py

```diff
@@ -152,30 +157,68 @@ async def create_sql_exporter_config(
         )
 
 
-def get_statefulset_affinity(name: str, logger: logging.Logger) -> Optional[V1Affinity]:
+def get_statefulset_affinity(
+    name: str, logger: logging.Logger, node_spec: Dict[str, Any]
+) -> Optional[V1Affinity]:
     if config.TESTING:
         logger.warning("Deploying cluster %s without any pod anti-affinity!", name)
         return None
 
-    return V1Affinity(
-        pod_anti_affinity=V1PodAntiAffinity(
-            required_during_scheduling_ignored_during_execution=[
-                V1PodAffinityTerm(
-                    label_selector=V1LabelSelector(
-                        match_expressions=[
-                            V1LabelSelectorRequirement(
-                                key=LABEL_COMPONENT, operator="In", values=["cratedb"]
-                            ),
-                            V1LabelSelectorRequirement(
-                                key=LABEL_NAME, operator="In", values=[name]
-                            ),
-                        ],
+    if is_shared_resources_cluster(node_spec):
+        return V1Affinity(
+            node_affinity=V1NodeAffinity(
+                required_during_scheduling_ignored_during_execution=V1NodeSelector(
+                    node_selector_terms=[
+                        V1NodeSelectorTerm(
+                            match_expressions=[
+                                V1NodeSelectorRequirement(
+                                    key="cratedb", operator="In", values=["shared"]
```

Comment:
I think it would be good if these were constants somewhere, just like we have it in the `V1Affinity` section below.